static/docs/user-guide/troubleshooting.md

```diff
@@ -0,0 +1,17 @@
+# Troubleshooting
+
+In this section we provide help for some of the problems that DVC user might
+stumbe upon.
+
+## Too many open files error
+
+A known problem some users run into with the `dvc pull`, `dvc fetch` and
+`dvc push` commands is `[Errno 24] Too many open files` (most common for S3
```

Comment:
We're still using this `[Errno 24] Too many open files` known issue as the first "trouble". All for including this section but this is not a good example as discussed previously :/ Sorry to go in circles!

Reply:
Maybe just rephrase along the lines of Ruslan's https://github.com/iterative/dvc.org/pull/875#issuecomment-568251704. Or perhaps @shcheklein has an idea of a good first, actual trouble to put here.

Reply:
@jorgeorpinel I think Ruslan's comment was specifically about this error, though. What is the GH ticket you are referring too?

My 2cs on this specific case. I agree with @efiop . It's impossible to solve `Too many open files` in general. This error in most cases means that some configuration is needed. It's very common for databases, for example, or other system level software. So, it can be part of the troubleshooting section.

I would say though, that I don't quite understand why do we use Errno 24 code - we should probably introduce our own codes for these situations or normal human readable messages with a link to this guide. But this is an UI/UX issue.

Reply:
> What is the GH ticket you are referring too?

https://github.com/iterative/dvc/issues/2473

I also agree with Ruslan but I though we had concluded that the `Too many open files` thing could possibly be fixed in the core code. If it has no solution, then this is OK. Up to you guys, feel free to mark this one resolved.

Reply:
Yep, that ticket is closed. Resolving this. @efiop @pared please, create an UI ticket/checkbox to improve the DVC core message (at lease include a link to this guide, may be introduce error.dvc.org the same way we have man.dvc.org to make it look short and nice and have a better level of control).

Reply:
@pared so, how does exactly the message look like now with the latest PR applied?

Reply:
@efiop @pared how about we do this properly? introduce our error codes? proper link to the section? hide details into Errno 24 into details (or even when you run it with `-v`)?

Reply:
@shcheklein 
This is how it looks like on current master:
https://asciinema.org/a/291591
and this is how it will look like after https://github.com/iterative/dvc/pull/3017:
https://asciinema.org/a/291592

Reply:
So, I still don't understand why do we need this part: 

`[Errno 24] Too many open files: '/home/prd/development/projects/playground/2473/repo/.dvc/cache/10/788f0c3b6f08ef6f183f49490183d0'`

we can keep it as a `-v` detail. We already have `Too many files message before`, why duplicate it? 

That path - looks ugly. I have no idea what does it mean, what operation it failed with? How can it help me?

Second problem - as I suggested we need to come up with our own links similar to `man.dvc.org`. `https://dvc.org/doc/user-guide/troubleshooting#too-many-open-files-error` looks ugly and a burden for us to support (let's imagine we decide to move/rename it?).

Let's discuss and create our codes and URLs like `https://error.dvc.org/dvc0001` or `https://error.dvc.org/too-many-open-files`?

Reply:
> we can keep it as a -v detail. We already have Too many files message before, why duplicate it?

That is how error chaining is implemented right now. `logger.exception` will automatically add the last exception so that you get some additional context. If we use `logger.error` instead, it will not show the traceback in the verbose mode, which will be harmful. To adjust that we will need to introduce a special extra flag for the log record, that would tell our logger to not append the error message but still append the stack. Or we could use `logger.error("...", stack_info=True)`, which is actually kinda meant for that :slightly_smiling_face: Created https://github.com/iterative/dvc/pull/3058 for that.

> Second problem - as I suggested we need to come up with our own links similar to man.dvc.org. https://dvc.org/doc/user-guide/troubleshooting#too-many-open-files-error looks ugly and a burden for us to support (let's imagine we decide to move/rename it?).
> Let's discuss and create our codes and URLs like https://error.dvc.org/dvc0001 or https://error.dvc.org/too-many-open-files?

Total overkill :smile:

Reply:
> Total overkill 😄

it's not, it's hard to manage direct links and it would take 10 minutes to setup it right.

Reply:
@shcheklein Got it. My thinking was that there is not enough links in use to bother with it, but I guess it is better to do sooner than later.

Reply:
@shcheklein @efiop 
So I should prepare some helper method, similar to `append_doc_link`?

Reply:
yep, that would be great ... and I can help to setup the `error.dvc.org` domain and redirects. The only question left - how do we determine the suffix/"error codes"?

Reply:
@shcheklein error codes are future-proof and easy to manage, but sometimes look uglier than text-based.Thinking about it, codes do seem pretty nice overall.

Reply:
Ok, so we do agree to set up the codes right? How do we assign them? Just as they occur? so this one will be 1? @efiop? @shcheklein I will definitely need help with setting up `error.dvc.org`

Reply:
@pared yep, you just agree with the team on the names (I don't have any strong preferences - just to come up with a nice and simple and stable URL). I'll setup `error.dvc.org` and redirects - np with that.

Reply:
Commented about this on https://github.com/iterative/dvc/pull/3017#pullrequestreview-339988084. Resolving